crates/oxc_parser/src/ts/statement.rs

```diff
@@ -357,75 +340,84 @@ impl<'a> Parser<'a> {
     }
 
     pub fn parse_ts_declaration_statement(&mut self) -> Result<Statement<'a>> {
-        let declaration = self.parse_declaration_clause()?;
-        Ok(Statement::Declaration(declaration))
-    }
-
-    pub fn parse_declaration_clause(&mut self) -> Result<Declaration<'a>> {
-        let has_ambient = self.ctx.has_ambient();
+        let reserved_ctx = self.ctx;
+        let start_span = self.start_span();
+        let (flags, modifiers) = self.eat_modifiers_before_declaration();
+        let declare = flags.declare();
+        let r#async = flags.r#async();
+        self.ctx = self.ctx.and_ambient(declare).and_await(r#async);
 
-        let declare = self.eat(Kind::Declare);
-        if declare {
-            self.ctx = self.ctx.and_ambient(true);
-        }
+        let result = self.parse_declaration(start_span, modifiers);
```

Comment:
@Boshen Addressed your comment. I made sure every time modifiers are skipped, the span is preserved. Would you like to have another look? Thanks!